superset-frontend/src/dashboard/actions/sliceEntities.js

```diff
@@ -38,96 +44,142 @@ export function fetchAllSlicesFailed(error) {
-            'datasource_url',
-            'datasource_name_text',
-            'description_markeddown',
-            'description',
-            'id',
-            'params',
-            'slice_name',
-            'url',
-            'viz_type',
-          ],
-          filters: [{ col: 'owners', opr: 'rel_m_m', value: userId }],
-          page_size: FETCH_SLICES_PAGE_SIZE,
-          order_column: 'changed_on_delta_humanized',
-          order_direction: 'desc',
-        })}`,
-      })
-        .then(({ json }) => {
-          const slices = {};
-          let { result } = json;
-          // disable add filter_box viz to dashboard
-          if (excludeFilterBox) {
-            result = result.filter(slice => slice.viz_type !== 'filter_box');
-          }
-          result.forEach(slice => {
-            let form_data = JSON.parse(slice.params);
-            form_data = {
-              ...form_data,
-              // force using datasource stored in relational table prop
-              datasource:
-                getDatasourceParameter(
-                  slice.datasource_id,
-                  slice.datasource_type,
-                ) || form_data.datasource,
-            };
-            slices[slice.id] = {
-              slice_id: slice.id,
-              slice_url: slice.url,
-              slice_name: slice.slice_name,
-              form_data,
-              datasource_name: slice.datasource_name_text,
-              datasource_url: slice.datasource_url,
-              datasource_id: slice.datasource_id,
-              datasource_type: slice.datasource_type,
-              changed_on: new Date(slice.changed_on_utc).getTime(),
-              description: slice.description,
-              description_markdown: slice.description_markeddown,
-              viz_type: slice.viz_type,
-              modified: slice.changed_on_delta_humanized,
-              changed_on_humanized: slice.changed_on_delta_humanized,
-            };
-          });
-
-          return dispatch(setAllSlices(slices));
-        })
-        .catch(
-          errorResponse =>
-            console.log(errorResponse) ||
-            getClientErrorObject(errorResponse).then(({ error }) => {
-              dispatch(
-                fetchAllSlicesFailed(
-                  error || t('Could not fetch all saved charts'),
-                ),
-              );
-              dispatch(
-                addDangerToast(
-                  t('Sorry there was an error fetching saved charts: ') + error,
-                ),
-              );
-            }),
-        );
+      return fetchSlices(userId, excludeFilterBox, dispatch, undefined);
     }
 
     return dispatch(setAllSlices(sliceEntities.slices));
   };
 }
+
+export function fetchSortedSlices(
+  userId,
+  excludeFilterBox = false,
+  order_column,
+) {
+  return dispatch => {
+    dispatch(fetchAllSlicesStarted());
+    return fetchSlices(
+      userId,
+      excludeFilterBox,
+      dispatch,
+      undefined,
```

Comment:
If passed as an `Option object` there's no need to worry about order or sending undefined.